optuna_dashboard/_app.py

```diff
@@ -411,6 +419,34 @@ def save_study_note(study_id: int) -> BottleViewReturn:
         response.status = 204  # No content
         return {}
 
+    @app.post("/api/trials/<trial_id:int>/tell")
+    @json_api_view
+    def tell_trial(trial_id: int) -> BottleViewReturn:
+        state = str_to_trial_state.get(request.json.get("state"))
+        if state is None or not state.is_finished():
+            response.status = 400  # Bad request
+            return {"reason": "state must be either 'Complete', 'Pruned' or 'Fail'."}
```

Comment:
If we don't need pruning for the time being, we could choose to not support it, i.e. return early. The logic is quite different from `Study.tell` in Optuna, and it'd make tests simpler. Just an idea.